src/assets/CurrencyRateController.ts

```diff
@@ -195,16 +195,27 @@ export class CurrencyRateController extends BaseController<
       pendingCurrentCurrency,
       pendingNativeCurrency,
     } = this.state;
+
+    const conversionDate: number = Date.now() / 1000;
+    let conversionRate: number | null = null;
+    let usdConversionRate: number | null = null;
     try {
-      const {
-        conversionDate,
-        conversionRate,
-        usdConversionRate,
-      } = await this.fetchExchangeRate(
-        pendingCurrentCurrency || currentCurrency,
-        pendingNativeCurrency || nativeCurrency,
-        this.includeUsdRate,
-      );
+      if (
+        (pendingCurrentCurrency || currentCurrency) &&
+        (pendingNativeCurrency || nativeCurrency) &&
+        pendingNativeCurrency !== ''
```

Comment:
Is it possible for `pendingNativeCurrency` to be set to an empty string? :thinking:

Reply:
I guess we're not stopping it. Huh. We should probably throw an error in the empty case, but I imagine we need better form validation in our products before we do that. Ah well.

Reply:
Though, because an empty string is falsey, I think we can omit it here. It will get replaced by `nativeCurrency` if it's empty.

```suggestion
```

Reply:
Ah! But if `setNativeCurrency` is called with an empty string, it'll never actually get used because of the `pendingNativeCurrency || nativeCurrency` below. Uh-oh. That's not good - that'll result in the wrong conversion rate being obtained again.

I think we need to change all uses of `pendingNativeCurrency || nativeCurrency` to allow the use of an empty string, OR investigate both products and ensure an empty native currency isn't possible. And the same for the currentCurrency as well. Otherwise we'll set the wrong conversionRate here, even after this PR.

Reply:
@Gudahtt, so we definitely can have an empty string for nativeCurrency in both products, this was one of the original impetuses for this work: https://github.com/MetaMask/metamask-extension/issues/9492

I'm not sure I completely follow your logic on that last point. Are you saying we should set and use the empty string even though its falsy?

Reply:
Well, ideally we'd disallow empty strings completely and throw an error, and prevent it using form validation.

But if we want or have to delay that, setting it to an empty string is our only option if we want the controller state to make sense. Otherwise we'd flip back to the last currency used and show the wrong rate, which was the exact thing we're trying to prevent here.

Reply:
Well so if we didn't try to fetch the exchange rate in the case where either pending currencies are an empty string then the conversionRate would be set to null as the logic stands in this PR?

Reply:
That is true, but the native currency would remain set to the previous setting, and this controller polls. The next update would change the conversation rate according to the previous setting.

Reply:
ah ok. So it matters in the update method. That makes sense.

Reply:
Ah I thought about that and thought - though it should be obvious from the name - that the nullish coalecer would have the same issue of passing on falsy values. Will make these changes and push an update